apps/users/models.py

```diff
@@ -145,6 +152,18 @@ def create_superuser(self, username, email, password):
         GroupUser.objects.create(user=user, group=admins)
         return user
 
+    def _generate_username(self, seed):
+        log.info('Generating username for {}'.format(seed))
+        if self.model.objects.filter(username=seed).exists():
+            username = '{seed}-{num}'.format(
+                seed=seed, num=random.randint(1000, 9999))
+            log.warning('Username taken for {} trying {}'.format(
+                seed, username))
+            return username
```

Comment:
* I chose this range because it's pretty big so an unintentional collision is unlikely.
* [`EmailField`](https://docs.djangoproject.com/en/1.6/ref/models/fields/#emailfield) has a `max_length` of 75 and the username has a `max_length` of 255 so this won't go over the max field size.
* If the username with the seed exists it will try it with the 4 digit random suffix and if that exists it will blow up.
* I doubt that this will ever happen but if someone decides to set their username to a different person's email address I'd rather not prevent that person from being able to create an account. If someone decides to create 9000 accounts to block someone from signing up then we can come up with a better solution.

Essentially I wanted this to be constant time. I don't want to iterate through all of the "clashes" because someone could craft some malicious data to cause the server to run this code for a long time. The likelihood of this not finding a valid username with non-malicious input is pretty much 0. There's no reason for someone to create usernames like this unless they are being intentionally malicious.